### ISSUE TITLE
React MouseEvent Type is not generic errors

### ISSUE THREAD

**Issue by rpurushotma** [palantir/blueprint#306]
I get the following errors trying to use blueprintjs in my project with typescript 2.0.10 compiler. Are there any flags or anything that need to be set in tsconfig or anywhere to get blueprint to work?

```
ERROR in 
[path]/gui/node_modules/@blueprintjs/core/dist/components/tree/treeNode.d.ts
(47,35): error TS2315: Type 'MouseEvent' is not generic.

ERROR in 
[path]/node_modules/@blueprintjs/core/dist/components/tree/treeNode.d.ts
(48,38): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/tree/treeNode.d.ts
(49,41): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/tree/treeNode.d.ts
(50,36): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/tree/tree.d.ts
(4,81): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/tag/tag.d.ts
(8,20): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/slider.d.ts
(25,39): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/slider.d.ts
(26,39): error TS2315: Type 'TouchEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/handle.d.ts
(30,43): error TS2315: Type 'TouchEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/handle.d.ts
(31,47): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/handle.d.ts
(32,52): error TS2315: Type 'TouchEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/coreSlider.d.ts
(58,48): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/coreSlider.d.ts
(59,48): error TS2315: Type 'TouchEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/rangeSlider.d.ts
(23,39): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/slider/rangeSlider.d.ts
(24,39): error TS2315: Type 'TouchEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/forms/radioGroup.d.ts
(21,23): error TS2315: Type 'FormEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/forms/controls.d.ts
(13,16): error TS2315: Type 'FormEventHandler' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/context-menu/contextMenuTarget.d.ts
(3,26): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/overlay/overlay.d.ts
(54,22): error TS2315: Type 'SyntheticEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/alert/alert.d.ts
(31,18): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/components/alert/alert.d.ts
(35,18): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/common/props.d.ts
(25,23): error TS2315: Type 'MouseEvent' is not generic.

ERROR in [path]/gui/node_modules/@blueprintjs/core/dist/common/props.d.ts
(41,16): error TS2315: Type 'FormEventHandler' is not generic.
```

**Comment by giladgray** [comment]
you need the latest React typings.

**Comment by rpurushotma** [comment]
Ah, thanks!

**Comment by mdanka** [comment]
@giladgray I've just updated my react typings (on 0.14), `typings.json` shows `"react": "registry:dt/react#0.14.0+20161203201537",`, but I still get the above errors. Do you have any advice?

Thanks!
Miklos

**Comment by giladgray** [comment]
@mdanka sorry i missed your comment all those weeks ago! can you try again now? i suspect things have changed greatly. it looks like DT merged their `types-2.0` branch into master so there's probably a newer version.
